scripts/prepare-crds

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+set -e
+
+# Split the provided package into a charts and charts-crd package
+
+if [[ -z $1 ]]; then
+	echo "No directory provided to initialize charts-crd within"
+	exit 1
+fi
+
+f=$1
+
+if [[ -d ${f}/charts-crd ]]; then
+	rm -rf ${f}/charts-crd
+fi
+
+if ! [[ -d ${f}/charts ]]; then
+	echo "Could not find ${f}/charts"
+    exit 1
+fi
+
+if ! [[ -f ${f}/charts/Chart.yaml ]]; then
+	echo "Could not find ${f}/charts/Chart.yaml"
+	exit 1
+fi
+
+if ! [[ -d ${f}/charts/crds ]] || [[ $(ls ${f}/charts/crds | wc -l) -eq 0 ]]; then
+	echo "Chart does not have any crds within a crd/ directory"
+	exit 1
+fi
+
+# Create directory and move CRDs
+mkdir -p ${f}/charts-crd/templates
+mv ${f}/charts/crds/* ${f}/charts-crd/templates
+rm -rf ${f}/charts/crds
+
+# Collect information on chart
+name=$(cat ${f}/charts/Chart.yaml | yq r - 'name')
+api_version=$(cat ${f}/charts/Chart.yaml | yq r - 'apiVersion')
+chart_version=$(cat ${f}/charts/Chart.yaml | yq r - 'version')
+
+# Collect information on CRDs
+crd_apis=()
+crd_gvrs=()
+for crd_yaml in ${f}/charts-crd/templates/*; do
+	crd_type=$(yq r ${crd_yaml} 'spec.names.plural')
+	crd_group=$(yq r ${crd_yaml} 'spec.group')
+	crd_version=$(yq r ${crd_yaml} 'spec.version')
+	crd_kind=$(yq r ${crd_yaml} 'spec.names.kind')
+
+	crd_gvrs+=("${crd_type}.${crd_group}/${crd_version}")
+	crd_apis+=("${crd_group}/${crd_version}/${crd_kind}")
+done
+
+# Init Chart.yaml for CRD chart
+cat << EOF > ${f}/charts-crd/Chart.yaml
+apiVersion: ${api_version}
+version: ${chart_version}
+description: A Rancher chart that creates ${name} CRDs within a cluster.
+name: ${name}-crd
+type: application
+EOF
+
+# Add annotations to charts/Chart.yaml pointing to CRD chart
+# Only add one GVR as the annotation
+yq w -i ${f}/charts/Chart.yaml "annotations[catalog.cattle.io/auto-install-gvr]" "${crd_gvrs[0]}"
```

Comment:
If another CRD is added to the folder it could change which CRD is used, but I suppose as long as they change at the same time its ok because this should only be used at install time. Alternative would be to have user provide them.

Reply:
I think it makes more sense for the user to provide it. Updated the fields provided by the user if they want to use the flag to instead be the following:
```
createCRDChart:
  enabled: true
  providesGVR: prometheuses.monitoring.coreos.com/v1
```

This way the user has control over what the value of the annotation will be so that it doesn't change on an upgrade.

Reply:
Updated [the example repo with `rancher-monitoring` changes](https://github.com/aiyengar2/charts/tree/add_split_crd_flag_example/packages/rancher-monitoring) based on the new flag used

Reply:
Looks good, thanks.